cmd/clusterctl/client/client.go

```diff
@@ -28,8 +28,8 @@ type Client interface {
 	// GetProvidersConfig returns the list of providers configured for this instance of clusterctl.
 	GetProvidersConfig() ([]Provider, error)
 
-	// GetProviderComponents returns the provider components for a given provider, targetNamespace, watchingNamespace.
-	GetProviderComponents(provider string, providerType clusterctlv1.ProviderType, targetNameSpace, watchingNamespace string) (Components, error)
+	// GetProviderComponents returns the provider components for a given provider with options including targetNamespace, watchingNamespace.
+	GetProviderComponents(provider string, providerType clusterctlv1.ProviderType, options repository.ComponentsOptions) (Components, error)
```

Comment:
Is it possible to define an alias for `repository.ComponentsOptions` in `cmd/clusterctl/client/alias.go` so we are not forcing to import the `repository` package to the consumer

Such alias should be used in all client the public methods

Reply:
will add the alias